proc_RunUpdateStatistics.sql

```diff
@@ -0,0 +1,300 @@
+      INNER JOIN
+         @v_RowCountsTable AS r ON r.SchemaName = s.SchemaName AND r.TableName = s.TableName
+      -- Skip tables that have very few rows, as updated stats will matter much less
+      WHERE
+         r.TableRowCount >= @p_MinimumTableRowCountToUpdate
+
+   SET @v_OperationStopTime = GETDATE();
+
+   SELECT @v_QueriesExecuted = @v_QueriesExecuted + @v_NewLine + @v_NewLine + 'Joined stale statistics info with table row counts';
+
+   SELECT @v_EmailReport = @v_EmailReport + @v_NewLine + 'Finished querying for statistics in need of update. ' + 'Started At: ' + CAST( @v_OperationStartTime AS VARCHAR(20) ) + ' Ended At: ' + CAST( @v_OperationStopTime AS VARCHAR(20) ) + ' Took ' + CAST( DATEDIFF( SECOND, @v_OperationStartTime, @v_OperationStopTime ) AS VARCHAR(20) )  + ' seconds';
+
+   DECLARE
+      @v_CurSchema AS SYSNAME
+      ,@v_CurTable AS SYSNAME
+      ,@v_CurIndex AS SYSNAME
+      ,@v_CurStatsLastUpdatedTime AS DATETIME2(0)
+      ,@v_CurRowCountOnLastStatsUpdate AS BIGINT
+      ,@v_CurCurrentRowCount AS BIGINT
+      ,@v_Counter AS SMALLINT
+      ,@v_LastRow AS SMALLINT
+      ,@v_UpdateStatisticsCmd AS NVARCHAR(MAX)
+
+   SET @v_Counter = 1;
+   SELECT @v_LastRow = COUNT(1) FROM @v_StaleStatisticsInformationWithRowCountTable;
+
+   -- Loop through the statistics and force a full scan
+   WHILE ( @v_Counter <= @v_LastRow )
+   BEGIN
+      SELECT
+         @v_CurSchema = '[' + s.SchemaName + ']'
+         ,@v_CurTable = '[' + s.TableName + ']'
+         ,@v_CurIndex = '[' + s.IndexName + ']'
+         ,@v_CurStatsLastUpdatedTime = s.StatsLastUpdatedTime
+         ,@v_CurRowCountOnLastStatsUpdate = s.RowCountOnLastStatsUpdate
+         ,@v_CurCurrentRowCount = s.CurrentTableRowCount
+      FROM
+         @v_StaleStatisticsInformationWithRowCountTable AS s
+      WHERE
+         s.StatisticsMaintenanceId = @v_Counter;
+
+      BEGIN TRY
+         SET @v_OperationStartTime = GETDATE();
+         SET @v_UpdateStatisticsCmd = 'UPDATE STATISTICS ' + @v_CurSchema + '.' + @v_CurTable + ' ' + @v_CurIndex + ' WITH FULLSCAN';
+         EXECUTE sp_executesql @v_UpdateStatisticsCmd;
+         SET @v_OperationStopTime = GETDATE();
+
+         SELECT @v_QueriesExecuted = @v_QueriesExecuted + @v_NewLine + @v_UpdateStatisticsCmd;
+
+         SELECT @v_EmailReport = @v_EmailReport + @v_NewLine + 'Updated statistics with full scan: ' + @v_CurSchema + '.' + @v_CurTable + '.' + @v_CurIndex + @v_NewLine + '...was previously updated on ' + CAST( @v_CurStatsLastUpdatedTime AS VARCHAR(20) ) + ', table had ' + CAST( @v_CurRowCountOnLastStatsUpdate AS VARCHAR(20) ) + ' rows at time of last update, and table now has ' + CAST( @v_CurCurrentRowCount AS VARCHAR(20) ) + ' rows.';
+      END TRY
+      BEGIN CATCH
+         SELECT @v_EmailReport = @v_EmailReport + @v_NewLine + 'Failed to execute statistics full scan statement: ' + @v_NewLine + @v_UpdateStatisticsCmd + @v_NewLine + 'Due to exception: ' + @v_NewLine + ERROR_MESSAGE();
+      END CATCH
```

Comment:
Add `GOTO done;` before ending the catch

Reply:
Talked with Paul - reason we do not want this is because we do not want to abort the entire job if one fails.